### DISCUSSION TITLE
Distribute App for Android

### DISCUSSION THREAD

**Discussion by eclere** [unoplatform/uno#16918] [Q&A] [answered]
Hello,
I probably have the problem that an older bug is active again. https://developercommunity.visualstudio.com/t/Csproj-Editor-does-not-open-due-to-Andro/10333050
My error report is similar: System.ArgumentException: Expected 12 values for property MauiSharedRule::ApplicationTitle, but got 6. The property page may not be defined for all configurations.

I use Visual Studio Community 17.10.0, as well as Preview 17.11.0 Preview 1.0

So I can't make any project settings. I receive an *.aab file via the publish, but I need *.apk. According to the help, the format should be adjustable in the project properties. Since I cannot open these, my idea would be to make this setting directly in the XML of the project file. Unfortunately I could not find any documentation.

I would like to use ad-hoc.

Does anyone have a tip?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report.

The error you're seeing is [this one](https://developercommunity.visualstudio.com/t/Project-Properties-failing-with:-System/10655592), coming from Visual Studio. We're in contact with Microsoft to get this resolved.

For the publication, we're currently extending our documentation for publishing, but you can take a look at the changes in https://github.com/unoplatform/uno/pull/16903 in the meantime.

**Reply by eclere** [reply]
Thank you.
` dotnet publish -f net8.0-android -c Release -o ./publish`
That worked for me

**Reply by agneszitte** [reply]
Thank you for the feedback @eclere. 
The documentation is now available on the website as well now: https://platform.uno/docs/articles/uno-publishing-android.html

**Comment by trydalch** [comment]
I'm getting this error just trying to open the project properties. I'm in a brand new Uno project, just trying to test debugging in each platform. We're doing a quick POC to evaluate Uno. 

I've built and run the app for web, windows, and android. I was in the process of setting up fro iOS with hot restart and that's when I ran into this issue. 

Possibly related, but perhaps a separate issue: I can't run it for ios (keeps popping up the local device setup dialog). When I try to build it for iOS, it cancels saying that the Verification of the iOS environment is running. I've given it ample time, but it continues giving that message. 

I've restarted VS several times. Created a fresh project. Nothing has fixed this.

**Reply by trydalch** [reply]
I've also gotten a warning/error banners in visual studio about mismatched number of target frameworks.

**Reply by agneszitte** [reply]
@trydalch for the project properties, you can upvote https://developercommunity.visualstudio.com/t/Project-Properties-failing-with:-System/10655592 if it is not done alsready (Related issue: https://github.com/unoplatform/uno/issues/16490).

If you still have additional issues related to iOS or anything else don't hesitate to open separate discussions or issues so we can better help you please. Thanks a lot in advance!
